src/Raven.Server/Json/BlittableJsonTextWriterExtensions.cs

```diff
@@ -964,67 +967,58 @@ public static void WriteIncludedCounterNames(this AbstractBlittableJsonTextWrite
             writer.WriteComma();
 
             var results = (object)result.Results;
+            ValueTask<(long NumberOfResults, long TotalDocumentsSizeInBytes)> writeResultsAsync;
             if (results is List<Document> documents)
             {
                 writer.WritePropertyName(nameof(result.Results));
-                var writeDocuments = writer.WriteDocumentsAsync(context, documents, metadataOnly, token);
-                (numberOfResults, totalDocumentsSizeInBytes) = writeDocuments.IsCompletedSuccessfully 
-                    ? writeDocuments.Result 
-                    : await writeDocuments;
-                writer.WriteComma();
+                writeResultsAsync = writer.WriteDocumentsAsync(context, documents, metadataOnly, token);
             }
             else if (results is List<BlittableJsonReaderObject> objects)
             {
                 writer.WritePropertyName(nameof(result.Results));
-                var writeObjects = writer.WriteObjectsAsync(context, objects, token);
-                
-                (numberOfResults, totalDocumentsSizeInBytes) = writeObjects.IsCompletedSuccessfully 
-                    ? writeObjects.Result 
-                    : await writeObjects;
-                
-                writer.WriteComma();
+                writeResultsAsync = writer.WriteObjectsAsync(context, objects, token);
             }
             else if (results is List<FacetResult> facets)
             {
                 numberOfResults = facets.Count;
                 writer.WriteArray(context, nameof(result.Results), facets, (w, c, facet) => w.WriteFacetResult(c, facet));
-                writer.WriteComma();
-                var maybeFlush = writer.MaybeFlushAsync(token);
-                if (maybeFlush.IsCompletedSuccessfully == false)    
-                    await maybeFlush;
+                writeResultsAsync = ValueTask.FromResult<(long NumberOfResults, long TotalDocumentsSizeInBytes)>((numberOfResults, totalDocumentsSizeInBytes));
             }
             else if (results is List<SuggestionResult> suggestions)
             {
                 numberOfResults = suggestions.Count;
                 writer.WriteArray(context, nameof(result.Results), suggestions, (w, c, suggestion) => w.WriteSuggestionResult(c, suggestion));
-                writer.WriteComma();
-                var maybeFlush = writer.MaybeFlushAsync(token);
-                if (maybeFlush.IsCompletedSuccessfully == false)    
-                    await maybeFlush;
+                writeResultsAsync = ValueTask.FromResult<(long NumberOfResults, long TotalDocumentsSizeInBytes)>((numberOfResults, totalDocumentsSizeInBytes));
             }
             else
                 throw new NotSupportedException($"Cannot write query result of '{typeof(TResult)}' type in '{result.GetType()}'.");
 
+            (numberOfResults, totalDocumentsSizeInBytes) = writeResultsAsync.IsCompletedSuccessfully 
+                ? writeResultsAsync.Result 
```

Comment:
Similar to the above. Isn't it implemented by the compiler? Are there any gains here?